Core/Chat/TwitchChat.cs

```diff
@@ -0,0 +1,177 @@
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Core.Configuration;
+using Microsoft.Extensions.Logging;
+using NodaTime;
+using Persistence.Models;
+using Persistence.Repos;
+using TwitchLib.Client;
+using TwitchLib.Client.Events;
+using TwitchLib.Client.Models;
+using TwitchLib.Communication.Clients;
+using TwitchLib.Communication.Models;
+
+namespace Core.Chat
+{
+    public sealed class TwitchChat : IChat
+    {
+        public event EventHandler<MessageEventArgs> IncomingMessage = null!;
+
+        private readonly ILogger<TwitchChat> _logger;
+        private readonly IClock _clock;
+        private readonly string _ircChannel;
+        private readonly ImmutableHashSet<IrcConfig.SuppressionType> _suppressions;
+        private readonly ImmutableHashSet<string> _suppressionOverrides;
+        private readonly IUserRepo _userRepo;
+        private readonly TwitchClient _twitchClient;
+
+        private bool _connected = false;
+
+        public TwitchChat(
+            ILoggerFactory loggerFactory,
+            IClock clock,
+            IrcConfig ircConfig,
+            IUserRepo userRepo)
+        {
+            _logger = loggerFactory.CreateLogger<TwitchChat>();
+            _clock = clock;
+            _ircChannel = ircConfig.Channel;
+            _suppressions = ircConfig.Suppressions;
+            _suppressionOverrides = ircConfig.SuppressionOverrides
+                .Select(s => s.ToLowerInvariant()).ToImmutableHashSet();
+            _userRepo = userRepo;
+
+            _twitchClient = new TwitchClient(
+                client: new TcpClient(new ClientOptions()),
+                logger: loggerFactory.CreateLogger<TwitchClient>());
+            var credentials = new ConnectionCredentials(
+                twitchUsername: ircConfig.Username,
+                twitchOAuth: ircConfig.Password);
+            _twitchClient.Initialize(
+                credentials: credentials,
+                channel: ircConfig.Channel,
+                // disable TwitchLib's command features, we do that ourselves
+                chatCommandIdentifier: '\0',
+                whisperCommandIdentifier: '\0');
+        }
+
+        public Task SendMessage(string message)
+        {
+            if (_suppressions.Contains(IrcConfig.SuppressionType.Message) &&
+                !_suppressionOverrides.Contains(_ircChannel))
+            {
+                _logger.LogDebug($"(suppressed) >#{_ircChannel}: {message}");
+                return Task.CompletedTask;
+            }
+            _logger.LogDebug($">#{_ircChannel}: {message}");
+            _twitchClient.SendMessage(_ircChannel, message);
```

Comment:
This will eventually need to be modified to properly support long messages. Probably just cutting it into multiple messages. Right now, twitchlib just drops too long messages with an error log